### PR TITLE
Fix iscsi volume list indexerror

### PR DESCRIPTION
**Pull request by Yingshun**
If the command "virsh vol-list" does not return a volume,
an indexError is reported. To avoid this error, add wait_for in
this PR.

Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

**Comment by Yingshun** [comment]
**Test result:**
```
(.libvirt-ci-venv-ci-runtest-ZOezU6) [root@yicui-host1 volume]# avocado run --vt-type libvirt --vt-machine-type q35 virsh.vol_download_upload.only_download.iscsi_pool.no_options --vt-connect-uri qemu:///system
JOB ID     : 7505986c788763ab5fcd511c41e2e71587b004fc
JOB LOG    : /root/avocado/job-results/job-2020-05-25T23.36-7505986/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.vol_download_upload.only_download.iscsi_pool.no_options: PASS (15.44 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 16.94 s
(.libvirt-ci-venv-ci-runtest-ZOezU6) [root@yicui-host1 volume]# avocado run --vt-type libvirt --vt-machine-type q35 virsh.vol_download_upload.mix_download_upload.dir_pool.sparse --vt-connect-uri qemu:///system
JOB ID     : ab5ee2a22220794ef0d9cd2431e478f1cbecebc6
JOB LOG    : /root/avocado/job-results/job-2020-05-25T23.38-ab5ee2a/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.vol_download_upload.mix_download_upload.dir_pool.sparse: PASS (44.76 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 46.23 s
(.libvirt-ci-venv-ci-runtest-ZOezU6) [root@yicui-host1 volume]# 
```

**Comment by Yingshun** [comment]
@jgao12345

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py [hunk 1]
```diff
@@ -136,6 +136,19 @@ def write_disk(test, guest_vm, target_name, size=200):
...
+    """
+    Wrapper for getting volume
+
+    :params pool_name: The name of pool
+    :return: First volume in volume list by pool
+    """
+    virsh.pool_refresh(pool_name, debug=True)
+    vol_list = virsh.vol_list(pool_name, debug=True).stdout_text.splitlines()
+    if len(vol_list) > 3:
+        return vol_list[2]
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py:149]: a little confused here, why vol_list[2] is  First volume in volume list?

Comment (Yingshun) [reply libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py:149]: The output contains standard headers, so it needs to skip them

Comment (chunfuwen) [reply libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py:149]: please check whether get_vol_list in libvirt.py can satisfy your requirement

Comment (Yingshun) [reply libvirt/tests/src/virsh_cmd/volume/virsh_vol_download_upload.py:149]: only checking vol_list does not work in this case. So try to add pool refresh before vol_list to see if this issue can be fixed.

**Comment by Yingshun** [comment]
@jgao12345  Thanks for your update.  
Closing it because there is a bug 1843791 in iscsi-initiator-utils.

**Pull request closed without merging**